packages/user-mgnt/src/features/getAvatar/handler.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors. OpenCRVS and the OpenCRVS
+ * graphic logo are (registered/a) trademark(s) of Plan International.
+ */
+import * as Hapi from '@hapi/hapi'
+import * as Joi from 'joi'
+import { unauthorized } from '@hapi/boom'
+import User, { IUserModel } from '@user-mgnt/model/user'
+import { AVATAR_API } from '@user-mgnt/constants'
+import fetch from 'node-fetch'
+
+interface IAvatarPayload {
+  userId: string
+}
+
+export default async function getUserAvatar(
+  request: Hapi.Request,
+  h: Hapi.ResponseToolkit
+) {
+  const { userId } = request.payload as IAvatarPayload
+
+  // tslint:disable-next-line
+  const user: IUserModel | null = await User.findById(userId)
+
+  if (!user) {
+    // Don't return a 404 as this gives away that this user account exists
```

Comment:
Can you elaborate on this a little bit? Wouldn't someone requesting this endpoint know the user exists also by not getting an `unauthorized` response? I see this comment many places in our codebase, but I'm not sure if this really applies here

Reply:
In our database, we don't have an avatar for all the users so we have to check if the user has avatar in DB otherwise we get it from [AVATAR_API](https://eu.ui-avatars.com/api/?name=) using their name stored in DB and also as this public endpoint anyone can hit this endpoint with random userId it will try to check if the user has an avatar in DB and occurred internal error.

Reply:
So if I understand correctly, now if an attacker would want to know whether a user exists by using this API endpoint:

- They **first make a request to some other endpoint** with their token and get a 200 response. Now they know their token shouldn't return unauthorized
- They query this endpoint with `userId: 123123123`, if they get
  - `unauthorized`, they know the user doesn't exist
  - **a generated avatar**, they know the user exists but has not changed their avatar
  - **user set avatar**, they know user exists and has set an avatar
  
  So effectively it's the same as returning a 404 response, right?

Reply:
I think as with this endpoint we are using, not usernames (that are easy to guess), but generated user ids (that are difficult to guess), we can safely just return 404 when the user doesn't exist. Also as this is an internal API that we expose through gateway, we should do this kind of "hiding of internal logic" in the gateway so our internal API stays as standard as possible

Reply:
Yeah. Agreed.  I have returned 404 from this endpoint.